### ISSUE TITLE
Quartz scheduler dependency conflicts on HikariCP

### ISSUE THREAD

**Issue by mgargadennec** [spring-projects/spring-boot#12364]
Quartz scheduler, in `spring-boot-starter-quartz`, has a compile dependency on `com.zaxxer:HikariCP-java6:jar:2.3.13:compile`  whereas spring-boot depends on `com.zaxxer:HikariCP:jar:2.7.8:compile`

Moreover Quartz seems to depend on C3P0 too.

Thoses dependencies should be excluded.

This can cause a `java.lang.NoSuchMethodError:` exception when Tomcat is loading the classes at startup : 

```
SEVERE: The required Server component failed to start so Tomcat is unable to start.
Caused by: java.lang.NoSuchMethodError: com.zaxxer.hikari.HikariDataSource.getMetricsTrackerFactory()Lcom/zaxxer/hikari/metrics/MetricsTrackerFactor
```

**Comment by ShortyFR** [comment]
Used this workaround for the moment on my new project : 

```
    <dependency>
      <groupId>com.blossom-project</groupId>
      <artifactId>blossom-starter-basic</artifactId>
      <version>${blossom.version}</version>
      <exclusions>
        <exclusion>
          <groupId>com.zaxxer</groupId>
          <artifactId>HikariCP-java6</artifactId>
        </exclusion>
      </exclusions>
    </dependency>
```

**Comment by xRomZak** [comment]
I've faced a similar issue after upgrading to Spring Boot 2.0.9, but this time with Quartz's HikariCP-java7 dependency

**Comment by snicoll** [comment]
@xRomZak thanks for the feedback. Can you please open a new issue with a bit more details?

**Comment by thackel** [comment]
This morning after the upgrade to Spring Boot 2.1.4 i experienced the same problem.
The duplicate class checker plugin stopped the build because of
`com.zaxxer:HikariCP:jar:3.2.0` vs `com.zaxxer:HikariCP-java7:jar:2.4.13`

I had to exclude the dependency (introduced by quartz 2.4.1) from the starter:
```
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-quartz</artifactId>
			<exclusions>
				<exclusion>
					<groupId>com.zaxxer</groupId>
					<artifactId>HikariCP-java7</artifactId>
				</exclusion>
			</exclusions>
		</dependency>
```
